opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/service/protocol/pis/HbciPayment.java

```diff
@@ -0,0 +1,99 @@
+package de.adorsys.opba.protocol.hbci.service.protocol.pis;
+
+import de.adorsys.multibanking.domain.Bank;
+import de.adorsys.multibanking.domain.BankAccess;
+import de.adorsys.multibanking.domain.BankAccount;
+import de.adorsys.multibanking.domain.BankApiUser;
+import de.adorsys.multibanking.domain.request.TransactionRequest;
+import de.adorsys.multibanking.domain.response.PaymentResponse;
+import de.adorsys.multibanking.domain.transaction.AbstractPayment;
+import de.adorsys.multibanking.domain.transaction.SinglePayment;
+import de.adorsys.multibanking.hbci.model.HbciConsent;
+import de.adorsys.opba.protocol.bpmnshared.service.context.ContextUtil;
+import de.adorsys.opba.protocol.bpmnshared.service.exec.ValidatedExecution;
+import de.adorsys.opba.protocol.hbci.context.HbciContext;
+import de.adorsys.opba.protocol.hbci.context.PaymentHbciContext;
+import de.adorsys.opba.protocol.hbci.service.protocol.CustomizedOnlineBankingService;
+import de.adorsys.opba.protocol.hbci.service.protocol.pis.dto.PaymentInitiateBody;
+import de.adorsys.opba.protocol.hbci.service.protocol.pis.dto.PisSinglePaymentResult;
+import lombok.RequiredArgsConstructor;
+import org.flowable.engine.delegate.DelegateExecution;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.springframework.stereotype.Service;
+
+import static de.adorsys.opba.protocol.bpmnshared.GlobalConst.SPRING_KEYWORD;
+import static de.adorsys.opba.protocol.hbci.constant.GlobalConst.HBCI_MAPPERS_PACKAGE;
+
+@Service("hbciPaymentExecutor")
+@RequiredArgsConstructor
+public class HbciPayment extends ValidatedExecution<PaymentHbciContext> {
+    private final CustomizedOnlineBankingService onlineBankingService;
+    private final PaymentMapper paymentMapper;
+
+    @Override
+    protected void doRealExecution(DelegateExecution execution, PaymentHbciContext context) {
+        HbciConsent consent = context.getHbciDialogConsent();
+        SinglePayment singlePayment = paymentMapper.map(context.getPayment());
+        BankAccount account = new BankAccount();
+        account.setIban(context.getAccountIban());
+        singlePayment.setPsuAccount(account);
+
+        TransactionRequest<SinglePayment> request = create(singlePayment, new BankApiUser(), new BankAccess(),
+                context.getBank(), consent);
+        PaymentResponse response = onlineBankingService.executePayment(request, String.valueOf(System.currentTimeMillis()));
```

Comment:
As E2E id didn't help - suggest to drop this `String.valueOf(System.currentTimeMillis())`